src/main/java/org/generationcp/middleware/dao/oms/CVTermDao.java

```diff
@@ -1889,4 +1780,89 @@ protected String convertDataTypeToVariableDtoScale(final Integer dataTypeId) {
+				+ "   AND (v.definition like :" + CVTermDao.FNAME
+				+ " 		or v.name like :" + CVTermDao.NAME
+				+ " 		or vpo.alias like :" + CVTermDao.ALIAS + " ) "//
+				+ " LIMIT 100 ");
+			sqlQuery.setParameter(CVTermDao.PROGRAMUUID, programUUID);
+			sqlQuery.setParameter(CVTermDao.FNAME, '%' + query + '%');
+			sqlQuery.setParameter(CVTermDao.NAME, '%' + query + '%');
+			sqlQuery.setParameter(CVTermDao.ALIAS, '%' + query + '%');
+
+			sqlQuery.setResultTransformer(Transformers.ALIAS_TO_ENTITY_MAP);
+			sqlQuery.addScalar(CVTermDao.VARIABLE_ID);
+			sqlQuery.addScalar(CVTermDao.VARIABLE_NAME);
+			sqlQuery.addScalar(CVTermDao.VARIABLE_DEFINITION);
+			sqlQuery.addScalar(CVTermDao.METHOD_ID);
+			sqlQuery.addScalar(CVTermDao.METHOD_NAME);
+			sqlQuery.addScalar(CVTermDao.METHOD_DEFINITION);
+			sqlQuery.addScalar(CVTermDao.PROPERTY_ID);
+			sqlQuery.addScalar(CVTermDao.PROPERTY_NAME);
+			sqlQuery.addScalar(CVTermDao.PROPERTY_DEFINITION);
+			sqlQuery.addScalar(CVTermDao.SCALE_ID);
+			sqlQuery.addScalar(CVTermDao.SCALE_NAME);
+			sqlQuery.addScalar(CVTermDao.SCALE_DEFINITION);
+			sqlQuery.addScalar(CVTermDao.VARIABLE_ALIAS);
+			sqlQuery.addScalar(CVTermDao.VARIABLE_EXPECTED_MIN);
+			sqlQuery.addScalar(CVTermDao.VARIABLE_EXPECTED_MAX);
+
+			final List<Map<String, Object>> queryResults = (List<Map<String, Object>>) sqlQuery.list();
+
+			final List<Variable> variables = new ArrayList<>();
+			for (final Map<String, Object> item : queryResults) {
+				final Variable variable =
+					new Variable(new Term(Util.typeSafeObjectToInteger(item.get(CVTermDao.VARIABLE_ID)), (String) item.get(CVTermDao.VARIABLE_NAME), (String) item.get(CVTermDao.VARIABLE_DEFINITION)));
+				variable.setMethod(new Method(new Term(Util.typeSafeObjectToInteger(item.get(CVTermDao.METHOD_ID)), (String) item.get(CVTermDao.METHOD_NAME), (String) item.get(CVTermDao.METHOD_DEFINITION))));
+				variable.setProperty(new org.generationcp.middleware.domain.ontology.Property(new Term(Util.typeSafeObjectToInteger(item.get(CVTermDao.PROPERTY_ID)), (String) item.get(CVTermDao.PROPERTY_NAME), (String) item.get(CVTermDao.PROPERTY_DEFINITION))));
+				variable.setScale(new org.generationcp.middleware.domain.ontology.Scale(new Term(Util.typeSafeObjectToInteger(item.get(CVTermDao.SCALE_ID)), (String) item.get(CVTermDao.SCALE_NAME), (String) item.get(CVTermDao.SCALE_DEFINITION))));
+
+				// Alias, Expected Min Value, Expected Max Value
+				final String pAlias = (String) item.get(CVTermDao.VARIABLE_ALIAS);
+				final String pExpMin = (String) item.get(CVTermDao.VARIABLE_EXPECTED_MIN);
+				final String pExpMax = (String) item.get(CVTermDao.VARIABLE_EXPECTED_MAX);
+
+				variable.setAlias(pAlias);
+				variable.setMinValue(pExpMin);
+				variable.setMaxValue(pExpMax);
+
+				variables.add(variable);
+			}
+
+			return variables;
+
+		} catch (final HibernateException e) {
+			throw new MiddlewareQueryException("Error with searchAttributes(query=" + query + "): " + e.getMessage(), e);
```

Comment:
method should be: `searchAttributeVariables`

Reply:
done.